OrbitLinuxTracing/PerfEventQueue.cpp

```diff
@@ -26,30 +23,76 @@ void PerfEventQueue::PushEvent(std::unique_ptr<PerfEvent> event) {
     const std::unique_ptr<std::queue<std::unique_ptr<PerfEvent>>>& queue = queue_it->second;
 
     queue->push(std::move(event));
-    queues_heap_.push(queue.get());
+    queues_heap_.emplace_back(queue.get());
+    MoveUpHeapBack();
   }
 }
 
 bool PerfEventQueue::HasEvent() { return !queues_heap_.empty(); }
 
-PerfEvent* PerfEventQueue::TopEvent() { return queues_heap_.top()->front().get(); }
+PerfEvent* PerfEventQueue::TopEvent() { return queues_heap_.front()->front().get(); }
 
 std::unique_ptr<PerfEvent> PerfEventQueue::PopEvent() {
-  std::queue<std::unique_ptr<PerfEvent>>* top_queue = queues_heap_.top();
-  queues_heap_.pop();
```

Comment:
I would really add a comment here why you don not pop the event queue here. I see that it was needed in the empty case anyways, but I feel a comment would help here.